src/Microsoft.AspNetCore.Mvc.Core/Internal/RedirectToRouteResultExecutor.cs

```diff
@@ -45,7 +46,24 @@ public void Execute(ActionContext context, RedirectToRouteResult result)
             }
 
             _logger.RedirectToRouteResultExecuting(destinationUrl, result.RouteName);
-            context.HttpContext.Response.Redirect(destinationUrl, result.Permanent);
+            if (result.PreserveMethod == false)
+            {
+                context.HttpContext.Response.Redirect(destinationUrl, result.Permanent);
+            }
+
+            else
+            {
+                if (result.Permanent)
+                {
+                    context.HttpContext.Response.StatusCode = 308;
```

Comment:
Use the constants from Http.Abstractions?
https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.AspNetCore.Http.Abstractions/StatusCodes.cs#L32